configure.ac

```diff
@@ -1303,6 +1303,77 @@
             [[#include <linux/net_tstamp.h>]])
     ])
 
+  # AF_XDP support
+    AC_ARG_ENABLE(af-xdp,
```

Comment:
I have taken you advice onboard and developed this, just wanted to hear your thoughts on the solution before committing to it. The documentation informs the user which libraries are required, and once installed it auto-enables ``AF_XDP`` during configuration :)

    `AC_ARG_ENABLE(af-xdp,
           AS_HELP_STRING([--enable-af-xdp], [Enable AF_XDP support [default=yes]]),
                        [enable_af_xdp=$enableval],[enable_af_xdp=yes])

    AS_IF([test "x$enable_af_xdp" = "xyes"], [
        # Check for the availability of elf
        AC_CHECK_LIB(elf,elf_begin,,[enable_af_xdp=no])

        # Conditionally check headers, only when found will it 'continue'
        AS_IF([test "x$enable_af_xdp" = "xyes"],
            # Check for the availability of libxdp
            AC_CHECK_HEADERS([xdp/xsk.h],,[enable_af_xdp=no])
            AC_CHECK_LIB([xdp],[xsk_umem__create],,[enable_af_xdp=no]))

        AS_IF([test "x$enable_af_xdp" = "xyes"],
            # Check for the availability of libbpf
            AC_CHECK_HEADERS([bpf/libbpf.h],,[enable_af_xdp=no])
            AC_CHECK_LIB([bpf],[bpf_object__open],,[enable_af_xdp=no]))

        # Are all required libs installed, yes=HAVE_AF_XDP
        AS_IF([test "x$enable_af_xdp" = "xyes"],
            AC_DEFINE([HAVE_AF_XDP],[1],[AF_XDP support is available]))
    ])`

Reply:
Is this going to be forgiving to users? Like not stopping with hard errors if some of the deps are missing?

Reply:
Yeah so basically its designed with one ``enable-af-xdp`` flag that is used to report ``yes|no``. If this is unset (when a library is not present) during configuration it wont even bother checking the next library dependancy (we want them all or its disabled). Finally, it gets to the end and as if its unset will not include this feature, otherwise its still true so its enabled.

Reply:
Great, thanks for confirming.